guppylang/definition/value.py

```diff
@@ -55,7 +54,7 @@ def synthesize_call(
         """Synthesizes the return type of a function call."""
 
     def __call__(self, *args: Any, **kwargs: Any) -> Any:
-        raise GuppyError("Guppy functions can only be called in a Guppy context")
+        raise RuntimeError("Guppy functions can only be called in a Guppy context")
```

Comment:
Guppy errors are now reserved for things that go wrong during compilation. This error is thrown immediately when the user calls the function, so we shouldn't use `GuppyError`